### PR TITLE
add notes about ACID transactions

### PR DESCRIPTION
**Pull request by DanRoscigno**
I made very few changes, the two that I recall are changing `max_insert_size` to `max_insert_block_size`, and changing the default from 1,000,000 rows to 1,048,576 rows.

### CODE REVIEW

File: docs/en/guides/developer/transactional.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Transactional (ACID) support
+
+INSERTs into one partition in one table of the MergeTree\* family up to `max_insert_block_size` rows are transactional (ACID):
+
+### Atomic
+
+INSERTs succeed or are rejected as a whole- if a confirmation is sent to the client, then all rows were inserted; if an error is sent to the client, then no rows were inserted.
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:7]:
```suggestion
INSERTs succeed or are rejected as a whole - if a confirmation is sent to the client, then all rows were inserted; if an error is sent to the client, then no rows were inserted.
```

File: docs/en/guides/developer/transactional.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Transactional (ACID) support
+
+INSERTs into one partition in one table of the MergeTree\* family up to `max_insert_block_size` rows are transactional (ACID):
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:3]: Are you sure this formatting is enough to understand that next points are valid only if `up to `max_insert_block_size` rows` is true? 
Actually for ACI. D is valid in any case.

File: docs/en/guides/developer/transactional.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Isolated
+
+Clients observe a consistent snapshot of the table - either the state of the table before an INSERT, or the state after a successful INSERT; no partial state is seen.
+
+### Durable
+
+A successful INSERT is written to the filesystem before sending a confirmation to the client, on single replica or multiple replica systems (this is controlled by the `insert_quorum` setting), and ClickHouse can ask the operating system to sync the filesystem data on the storage media (this is controlled by the `fsync_after_insert` setting).
+
+### Details
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:21]: I would not add it on the same level as ACID headers. Probably converting previous headers to a table or just a list will make it less tricky.

File: docs/en/guides/developer/transactional.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+- If a table has multiple partitions and the INSERT covers multiple partitions - then insertion into each partition is transactional on its own.
+
+- INSERTs into multiple tables with one statement is possible if materialized views are used.
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:25]: Probably something like that is more correct? But I'm not a native speaker.
```suggestion
- Atomic INSERT into multiple tables with one statement is possible if materialized views are used.
```
@alexey-milovidov How do you think, do we need to cover here what happens if MV insert fails and how deduplication works for MV?

```suggestion
- INSERTs into multiple tables with one statement are possible if materialized views are used.
```

File: docs/en/guides/developer/transactional.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+- Atomicity is ensured even if `async_insert` is enabled, but can be disabled with the `wait_for_async_insert` setting.
+
+- The `max_insert_block_size` is 1,048,576 rows by default and can be adjusted as needed.
+
+
+- ClickHouse uses multiversion concurrency control (MVCC) with snapshot isolation internally.
+
+- All ACID properties are valid even in the case of server kill / crash.
+
+- Using either insert_quorum into different availability zones or `fsync` should be enabled to ensure durable inserts in the typical setup.
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:40]:
```suggestion
- Using either `insert_quorum` into different availability zones or `fsync_after_insert=1` to ensure durable inserts in the typical setup.
```

File: docs/en/guides/developer/transactional.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+- "Consistency" in ACID terms does not cover the semantics of distributed systems, see https://jepsen.io/consistency which is controlled by different settings ([`select_sequential_consistency`](../../operations/settings/settings.md/#settings-select_sequential_consistency))
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:42]:
```suggestion
- "Consistency" in ACID terms does not cover the semantics of distributed systems, see [Jepsen](https://jepsen.io/consistency) which is controlled by different settings ([`select_sequential_consistency`](../../operations/settings/settings.md/#settings-select_sequential_consistency))
```

File: docs/en/guides/developer/transactional.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+- This explanation does not cover a new transactions feature that allows full-featured transactions for multiple SELECTS over multiple tables and materialized views, etc.
+
+- If a client does not receive response from the server, the client does not know if the transaction succeeded and it can repeat the transaction, using exactly-once insertion properties.
+:::note
+With `async_insert` deduplication will not work, so using retry will result in at least once transaction.
+:::
```
Comment (qoega) [inline docs/en/guides/developer/transactional.md:49]:
```suggestion
- With `async_insert` deduplication will not work per INSERT, so retries on failures will result in at least once semantics of INSERTs.
```

File: docs/en/guides/developer/transactional.md (hunk 2 above)
Comment (DanRoscigno) [reply docs/en/guides/developer/transactional.md]: Thanks @qoega , I updated, here is how it reads now.  Also moved the ACID lines from H2 to a dictionary:
image

File: docs/en/guides/developer/transactional.md (hunk 3 above)
Comment (DanRoscigno) [reply docs/en/guides/developer/transactional.md]: see above screen-grab

File: docs/en/guides/developer/transactional.md (hunk 4 above)
Comment (alexey-milovidov) [reply docs/en/guides/developer/transactional.md]: Let's merge, then improve...

**Pull request merged**